tika-parsers/tika-parsers-classic/tika-parsers-classic-modules/tika-parser-zip-commons/src/main/java/org/apache/tika/detect/zip/DefaultZipContainerDetector.java

```diff
@@ -241,12 +247,33 @@ private MediaType detect(ZipArchiveEntry zae, ZipArchiveInputStream zis,
     }
 
     private MediaType finalDetect(StreamingDetectContext detectContext) {
-        for (ZipContainerDetector d : zipDetectors) {
+        for (ZipContainerDetector d : getDetectors()) {
             MediaType mt = d.streamingDetectFinal(detectContext);
             if (mt != null) {
                 return mt;
             }
         }
         return MediaType.APPLICATION_ZIP;
     }
+
+    private List<ZipContainerDetector> getDetectors() {
+        List<ZipContainerDetector> zipDetectors = new ArrayList<>();
+        ServiceLoader loader = getServiceLoader();
+        zipDetectors.addAll(loader.loadDynamicServiceProviders(ZipContainerDetector.class));
+        if (staticZipDetectors0 == null) {
+            staticZipDetectors0 = new ArrayList<ZipContainerDetector>();
+            staticZipDetectors0.addAll(loader.loadStaticServiceProviders(ZipContainerDetector.class));
+        }
+        
+        zipDetectors.addAll(staticZipDetectors0);
+        return zipDetectors;
+    }
+
+    private ServiceLoader getServiceLoader() {
```

Comment:
I suggest to try to use the same pattern for getting the service loader as in all other classes.

Reply:
okay - just means a bit more complex getDetectors - because this class works differently as the other as it has constructor to directly set the "detectors" (but this could be null), and if they are set the loader is null